File: week6/whale.py
#!/usr/bin/python
import sys

pods = 0
count = 0

for line in sys.stdin.read().strip().split('\n') :
    arr = line.strip().split(' ', 2)
    if(arr[1] == sys.argv[1]) :
        pods += 1
        count += int(arr[0])

print( "%s observations: %d pods, %d individuals\n" % (sys.argv[1], pods, count))

File: week6/whale.pl
#!/usr/bin/perl
use warnings;
my $pods = 0;
my $count = 0;
my @arr = [];
while(<STDIN>) {
  chomp;
  @arr = split(/ /,$_,2);
  if ( $arr[1] eq  $ARGV[0] ) {
    $pods = $pods + 1;
    $count = $count + $arr[0];
  }
}
print "$ARGV[0] observations: $pods pods, $count individuals\n";
